File: data/EXP01/python/2410/2410.output_1.py
# @lc app=leetcode id=2410 lang=python3
class Solution:
    def matchPlayersAndTrainers(self, players, trainers):
        players.sort()
        trainers.sort()
        
        num_matchings = 0
        used_trainers = set()
        
        for player in players:
            for j, trainer in enumerate(trainers):
                if j not in used_trainers and trainer >= player:
                    num_matchings += 1
                    used_trainers.add(j)
                    break
        
        return num_matchings

File: data/EXP01/golang_no_system_var/2410/2410.output_1.go
func matchPlayersAndTrainers(players []int, trainers []int) int {
    sort.Ints(players)
    sort.Ints(trainers)

    playerIdx, trainerIdx := 0, 0
    matches := 0

    for playerIdx < len(players) && trainerIdx < len(trainers) {
        if players[playerIdx] <= trainers[trainerIdx] {
            playerIdx++
            trainerIdx++
            matches++
        } else {
            trainerIdx++
        }
    }

    return matches
}
